controller.py

```diff
@@ -236,14 +236,38 @@ def libraries(self, live=True):
         # Return a specific set of information about all libraries in production; this generates the library list in the admin interface.
         # If :param live is set to False, libraries in testing will also be shown.
         result = []
-        libraries = self._db.query(Library).order_by(Library.name)
+        alphabetical = self._db.query(Library).order_by(Library.name)
+
+        # Load all the ORM objects we'll need for these libraries in a single query.
+        alphabetical = alphabetical.options(
+            joinedload(Library.hyperlinks),
+            joinedload('hyperlinks', 'resource'),
+            joinedload('hyperlinks', 'resource', 'validation'),
+            joinedload(Library.service_areas),
+            joinedload('service_areas', 'place'),
+            joinedload('service_areas', 'place', 'parent'),
```

Comment:
We often need the parent of the library's service area when generating its human-readable name (e.g. the "IL" in "Cook County, IL").